tagstudio/src/qt/modals/fix_unlinked.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (C) 2024 Travis Abendshien (CyanVoxel).
+# Licensed under the GPL-3.0 License.
+# Created for TagStudio: https://github.com/CyanVoxel/TagStudio
+
+
+import logging
+import typing
+
+from PySide6.QtCore import QThread, Qt, QThreadPool
+from PySide6.QtWidgets import QWidget, QVBoxLayout, QHBoxLayout, QLabel, QPushButton
+
+from src.core.library import Library
+from src.qt.helpers import FunctionIterator, CustomRunnable
+from src.qt.modals import DeleteUnlinkedEntriesModal, RelinkUnlinkedEntries
+from src.qt.widgets	import ProgressWidget
+
+# Only import for type checking/autocompletion, will not be imported at runtime.
+if typing.TYPE_CHECKING:
+	from src.qt.ts_qt import QtDriver
+
+
+ERROR = f'[ERROR]'
+WARNING = f'[WARNING]'
+INFO = f'[INFO]'
+
+logging.basicConfig(format="%(message)s", level=logging.INFO)
+
+
+class FixUnlinkedEntriesModal(QWidget):
+	# done = Signal(int)
+	def __init__(self, library:'Library', driver:'QtDriver'):
+		super().__init__()
```

Comment:
Only side effect is that the type hint has to be a string type, this appears to be fairly well accepted though so I don't see an issue with this side effect.